litex_boards/targets/ulx4m_ld_v2.py

```diff
@@ -0,0 +1,251 @@
+                "MT41K128M16": MT41K128M16,
+                "MT41K256M16": MT41K256M16,
+                "MT41K512M16": MT41K256M16,
+            }
+            sdram_module = available_sdram_modules.get(sdram_device)
+
+            self.submodules.ddrphy = ECP5DDRPHY(
+                pads = PHYPadsReducer(platform.request("ddram"), [0, 1]),
+                sys_clk_freq=sys_clk_freq)
+            self.comb += self.crg.stop.eq(self.ddrphy.init.stop)
+            self.comb += self.crg.reset.eq(self.ddrphy.init.reset)
+            self.add_sdram("sdram",
+                phy           = self.ddrphy,
+                module        = sdram_module(sys_clk_freq, "1:4"),
+                l2_cache_size = kwargs.get("l2_size", 8192)
+            )
+
+        # Ethernet / Etherbone ---------------------------------------------------------------------
+        if with_ethernet or with_etherbone:
+            self.submodules.ethphy = LiteEthPHYRGMII(
+                clock_pads = self.platform.request("eth_clocks"),
+                pads       = self.platform.request("eth"))
+            if with_ethernet:
+                self.add_ethernet(phy=self.ethphy, dynamic_ip=eth_dynamic_ip)
+            if with_etherbone:
+                self.add_etherbone(phy=self.ethphy, ip_address=eth_ip)
+
+        # SPI Flash --------------------------------------------------------------------------------
+        if with_spi_flash:
+            from litespi.modules import IS25LP128
+            from litespi.opcodes import SpiNorFlashOpCodes as Codes
+            self.add_spi_flash(mode="4x", module=IS25LP128(Codes.READ_1_1_4))
+
+        # Video ------------------------------------------------------------------------------------
+        if with_video_terminal or with_video_framebuffer:
+            self.submodules.videophy = VideoHDMIPHY(platform.request("gpdi"), clock_domain="hdmi")
+            if with_video_terminal:
+                self.add_video_terminal(phy=self.videophy, timings="640x480@75Hz", clock_domain="hdmi")
+            if with_video_framebuffer:
+                self.add_video_framebuffer(phy=self.videophy, timings="640x480@75Hz", clock_domain="hdmi")
+
+        # Leds -------------------------------------------------------------------------------------
+        if with_led_chaser:
+            self.submodules.leds = LedChaser(
+                pads         = platform.request_all("user_led"),
+                sys_clk_freq = sys_clk_freq)
+
+
+        # GPIOs ------------------------------------------------------------------------------------
+        if with_syzygy_gpio:
+            platform.add_extension(ulx4m_ld_v2.raw_syzygy_io("SYZYGY0"))
+            self.submodules.gpio = GPIOTristate(platform.request("SYZYGY0"))
+
+# Build --------------------------------------------------------------------------------------------
+
+def main():
+    parser = argparse.ArgumentParser(description="LiteX SoC on ULX4M-LD-V2")
```

Comment:
Small remark: why not using `LiteXArgumentParser` as done for [radiona_ulx3s](https://github.com/litex-hub/litex-boards/blob/master/litex_boards/targets/radiona_ulx3s.py#L149).
This reduce (a bit) the list arguments to be explicitly added.

Reply:
Yes I also need to rename file - Sorry for problems I am doing editing directly as I have more not important changes locally, like v1 and SDRAM version. Will change as you suggested...

Reply:
@trabucayre do you thing it is ok now?